src/proof.rs

```diff
@@ -1,22 +1,20 @@
-use crate::hash::{Algorithm, Hashable};
-use crate::merkle::get_merkle_proof_lemma_len;
-#[cfg(test)]
-use crate::merkle::Element;
-
 use anyhow::Result;
 use std::marker::PhantomData;
 use typenum::marker_traits::Unsigned;
 use typenum::U2;
 
+use crate::hash::{Algorithm, Hashable};
+use crate::merkle::get_merkle_proof_lemma_len;
 #[cfg(test)]
-use typenum::{U0, U1, U3, U4, U5, U8};
-
+use crate::merkle::Element;
 #[cfg(test)]
 use crate::merkle::MerkleTree;
 #[cfg(test)]
 use crate::store::VecStore;
 #[cfg(test)]
 use crate::test_common::{get_vec_tree_from_slice, Item, Sha256Hasher, XOR128};
+#[cfg(test)]
+use typenum::{U0, U1, U3, U4, U5, U8};
```

Comment:
If you have tests within a file, what you usually do is to put them into a `tests` module (see https://doc.rust-lang.org/book/ch11-03-test-organization.html#unit-tests for more information). This way you can import the things you need for testing there, and won't need to the `#[cfg(test)]`s for the imports.

So it would then look like this:

```rust
#[cfg(test)]
mod tests {
    use typenum::{U0, U1, U3, U4, U5, U8};

    fn modify_proof<…>(proof: &mut Proof<E, BaseTreeArity>) { … }

    #[test]
    fn test_proofs() {
    …
}
```

Though if the idea is to concentrate on the integration tests only, and you plan to clean those things up in a separate PR, that's also totally fine.